web/app/controller/Root.js

```diff
@@ -147,6 +148,12 @@ Ext.define('Traccar.controller.Root', {
                             text = typeKey;
                         }
                     }
+                    if (array[i].geofenceId !== 0) {
+                        geofence = Ext.getStore('Geofences').getById(array[i].geofenceId);
```

Comment:
Is this global variable? There should be no globals.

Also, have you run `jshist` and `jscs` on the code?